src/SonarScanner.MSBuild.PreProcessor/JreCaching/JreCache.cs

```diff
@@ -141,7 +141,27 @@ private static void EnsureClosed(Stream fileStream)
 
     private async Task<JreCacheResult> UnpackJre(string downloadTarget, JreDescriptor jreDescriptor, string cacheRootLocation)
     {
-        return new JreCacheFailure("NotImplemented. The JRE is downloaded, but we still need to check, unpack, and set permissions.");
+        if (ValidateChecksum(downloadTarget, jreDescriptor.Sha256))
+        {
+            return new JreCacheFailure("NotImplemented. The JRE is downloaded and validated, but we still need to unpack, and set permissions.");
+        }
+        return new JreCacheFailure(Resources.ERR_JreChecksumMissmatch);
```

Comment:
// TODO: Remove the archive as it doesn't match the checksum. Otherwise, any retries will fail as well.